MSAL/src/MSALPublicClientApplication.m

```diff
@@ -116,16 +144,34 @@ - (id)initWithClientId:(NSString *)clientId
 
 - (NSArray <MSALUser *> *)users:(NSError * __autoreleasing *)error
 {
-    return [_tokenCache getUsers:self.clientId context:nil error:error];
+    NSMutableSet *users = [NSMutableSet new];
+    
+    __auto_type tokens = [self.tokenCache getAllClientRTs:self.clientId context:nil error:error];
+    for (MSIDRefreshToken *token in tokens)
+    {
+        MSALUser *user = [[MSALUser alloc] initWithIdToken:[[MSIDAADV2IdTokenWrapper alloc] initWithRawIdToken:token.idToken]
+                                                clientInfo:token.clientInfo environment:token.authority.msidHostWithPortIfNecessary];
+        
+         [users addObject:user];
+    }
+
+    return [users allObjects];
 }
 
 - (MSALUser *)userForIdentifier:(NSString *)identifier
                           error:(NSError * __autoreleasing *)error
 {
-    return [_tokenCache getUserForIdentifier:identifier
-                                    clientId:self.clientId
-                                 environment:[self.authority host]
-                                       error:error];
+    __auto_type users = [self users:error];
+    
+    for (MSALUser *user in users)
+    {
+        if ([user.userIdentifier isEqualToString:identifier])
```

Comment:
this should also check for environment and clientID

Reply:
done